smolyay/grid.py

```diff
@@ -2,348 +2,429 @@
-    indices are computed, which then can be
-    replaced with indexes based on the points' distribution at each
-    indicy, determined by levels (each indicy is the index of the levels,
-    extracted from ``basis_set``). Then, grid points' indexes are
-    generated by making all combinations of indexes for
-    each combination of levels.
-
-    Once grid points' indexes are generated, actual grid points
-    and their corresponding basis functions can be made by simply replacing
-    the indexes with their corresponding points and basis functions.
-    Multi-dimensional levels (levels' indexes each basis function was taken
-    from) are also generated.
-    All the mentioned outputs are stored in a :class:`NestedIndexGrid`
-    object.
 
     Parameters
     ----------
-    basis_set: :class:`NestedBasisFunctionSet`
-        Unidimensional points, basis functions and levels.
+    domain: list
+        domain of the random points
 
-    Raises
-    ------
-    TypeError
-        ``basis_set`` must be a :class:`NestedBasisFunctionSet`.
-
-    Example
-    -------
-    Consider dimension (n) = 2, and exactness (\mu) = 1 and
-    Chebyshev's polynomial of first kind, and its extremums as
-    the basis function and points, then:
-    ..math::
-        levels in one-dimension = [[0], [1,2]]
-        indices = [[1,1], [1,2], [2,1]]
-    replacing each indicy with corresponding indexes determined by levels:
-    ..math::
-        level's combination = [([0],[0]), ([0],[1,2]), ([1,2],[0])]
-        then:
-        grid points indexes = [(0,0), (0,1), (0,2), (1,0), (2,0)]
-        points in one-dimension = [0, -1, 1]
-        basis in one-dimension = [:class:`ChebyshevFirstKind`,
-                :class:`ChebyshevFirstKind`, :class:`ChebyshevFirstKind`]
-
-    Then:
-    ..math:
-        grid points = [(0, 0), (0, -1), (0, 1), (-1, 0), (1, 0)]
-        levels in two dimension: [(0, 0), (0, 1), (0, 1), (1, 0), (1, 0)]
-        grid points' functions = [(:class:`ChebyshevFunction`,
-        :class:`ChebyshevFunction`)
-        , (:class:`ChebyshevFirstKind`, :class:`ChebyshevFirstKind`),
-        (:class:`ChebyshevFirstKind`, :class:`ChebyshevFirstKind`),
-        (:class:`ChebyshevFirstKind`, :class:`ChebyshevFirstKind`),
-        (:class:`ChebyshevFirstKind`, :class:`ChebyshevFirstKind`)]
+    number_points: int
+        number of points to generate
+
+    method: {"uniform", "halton", "sobol", "latin"}
+        the method of generating random points
+
+    seed: {int, numpy.random.Generator}
+        seed for generating the random points
 
+    options: {None, dict}, optional
+        if method != None, additional parameters passed to the QMCEngine
     """
 
-    def __init__(self, basis_set):
-        super().__init__(basis_set)
-        if not isinstance(self.basis_set, NestedBasisFunctionSet):
-            raise TypeError('Basis set must be nested for Smolyak grid')
+    def __init__(self, domain, number_points, method, seed, options=None):
```

Comment:
Each of these `method` options should be its own class, that takes its options as explicit constructor arguments / makes them available as properties. You should still use a base `RandomPointSet` that supplies the common parts for all of them. For example,

```py
class UniformRandomPointSet(RandomPointSet):
    ...
```